xbmc/PlayListPlayer.cpp

```diff
@@ -362,6 +363,17 @@ int CPlayListPlayer::GetCurrentPlaylist() const
   return m_iCurrentPlayList;
 }
 
+int CPlayListPlayer::GetCurrentPlaylist(int iPlaylist) const
+{
+  if (GetCurrentPlaylist() != PLAYLIST_NONE)
+    return GetCurrentPlaylist();
+
+  if (g_application.m_pPlayer->GetPreferredPlaylist() != PLAYLIST_NONE)
+    return g_application.m_pPlayer->GetPreferredPlaylist();
```

Comment:
The compiler should optimize it. It seems to be easier to read than adding a temp variable, but no problem changing it if that's the consensus.